### PR TITLE
[HIP] [DCP] Enable fused indexer QK preparation

### PR DESCRIPTION
**Pull request by ZhiweiYan-96**
## Summary
Enable the fused indexer QK preparation kernel for DCP. On non-owner DCP ranks, `slot_mapping` is negative, but those ranks must still produce valid `q_out` and `weights_out` for distributed index scoring. The previous early return skipped the entire kernel and left these outputs uninitialized.

## Changes
- Add an optional `compute_all_q_rope` argument, following the approach used by #4342.
- Compute Q RoPE, FP8 quantization, and `weights_out` on every DCP rank.
- Keep K normalization, quantization, and cache writes restricted to valid owner slots.
- Clamp stale padding positions before indexing the RoPE cache.
- Preserve the existing non-DCP behavior by default.
- Add coverage for mixed valid/invalid slots, stale positions, and the default non-DCP path.

### CODE REVIEW

File: csrc/kernels/cache_kernels.cu [hunk 1]
```diff
@@ -1451,10 +1453,10 @@ __global__ void indexer_qk_rope_quant_and_cache_kernel(
         return;
 
     const int64_t slot_idx = slot_mapping[token_idx];
-    if(slot_idx < 0)
+    if(!compute_all_q_rope && slot_idx < 0)
         return;
-
-    const int64_t pos = positions[token_idx];
+    int64_t pos = positions[token_idx];
+    pos = pos < 0 ? 0 : (pos >= max_position ? max_position - 1 : pos);
```
Comment (yitingw1) [inline csrc/kernels/cache_kernels.cu]: The clamp sits before the head_idx != 0 || slot_idx < 0 return, so it also applies to valid owner rows on the pre-existing non-DCP path.

File: csrc/kernels/cache_kernels.cu [hunk 2]
```diff
@@ -4102,6 +4106,7 @@ void indexer_qk_rope_quant_and_cache(
     AITER_CHECK(cos_cache.size(0) == sin_cache.size(0) &&
                     cos_cache.size(1) == sin_cache.size(1),
                 "cos_cache and sin_cache shapes must match");
+    AITER_CHECK(max_position > 0, "cos_cache and sin_cache must not be empty");
```
Comment (yitingw1) [inline csrc/kernels/cache_kernels.cu:4112]: Maybe add `AITER_CHECK(!compute_all_q_rope || q.size(0) == num_tokens,
            "compute_all_q_rope requires q to have exactly num_tokens rows");`, too?

File: csrc/kernels/cache_kernels.cu (hunk 1 above)
Comment (ZhiweiYan-96) [reply csrc/kernels/cache_kernels.cu]: The position clamp is now restricted to slot_idx < 0, so the existing valid-owner and non-DCP paths remain unchanged.

File: csrc/kernels/cache_kernels.cu (hunk 2 above)
Comment (ZhiweiYan-96) [reply csrc/kernels/cache_kernels.cu:4112]: added, thanks for reminding.

**Review by yitingw1 — DISMISSED** [review]
LGTM

**Review by XiaobingSuper — DISMISSED** [review]

File: op_tests/test_indexer_qk_rope_quant_and_cache.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
+# SPDX-License-Identifier: MIT
+
+import torch
+
+from aiter import dtypes, indexer_qk_rope_quant_and_cache
+
+
```
Comment (yzhou103) [inline op_tests/test_indexer_qk_rope_quant_and_cache.py:123]: it seems there's no test for indexer_qk_rope_quant_and_cache before. it's better add test with/without comupute_all_q_rope  and there is a skill in .claude about aiter-op test. you can refer to that.

**Review by valarLip — DISMISSED** [review]

File: op_tests/test_indexer_qk_rope_quant_and_cache.py (hunk 3 above)
Comment (ZhiweiYan-96) [reply op_tests/test_indexer_qk_rope_quant_and_cache.py:123]: image
case updated, thanks for reminding.

**Review by valarLip — APPROVED** [review]
